hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/CoalescedWriteBehindQueue.java

```diff
@@ -9,187 +9,145 @@
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
+
+import static com.hazelcast.util.CollectionUtil.isEmpty;
+import static com.hazelcast.util.ValidationUtil.checkNotNull;
 
 /**
  * A write-behind queue which supports write coalescing.
  */
 class CoalescedWriteBehindQueue implements WriteBehindQueue<DelayedEntry> {
 
-    protected Map<Data, DelayedEntry> queue;
+    protected Map<Data, DelayedEntry> map;
 
     public CoalescedWriteBehindQueue() {
-        queue = new LinkedHashMap<Data, DelayedEntry>();
-    }
-
-    public CoalescedWriteBehindQueue(Map<Data, DelayedEntry> queue) {
-        this.queue = queue;
+        map = new LinkedHashMap<Data, DelayedEntry>();
     }
 
     @Override
-    public boolean offer(DelayedEntry delayedEntry) {
+    public void addEnd(DelayedEntry delayedEntry) {
         if (delayedEntry == null) {
-            return false;
+            return;
         }
         calculateStoreTime(delayedEntry);
-        final Data key = (Data) delayedEntry.getKey();
-        queue.put(key, delayedEntry);
-        return true;
-    }
-
-    /**
-     * If this is an existing key in this queue, use previously set store time;
-     * since we do not want to shift store time of an existing key on every update.
-     */
-    private void calculateStoreTime(DelayedEntry delayedEntry) {
         Data key = (Data) delayedEntry.getKey();
-        DelayedEntry currentEntry = queue.get(key);
-        if (currentEntry != null) {
-            long currentStoreTime = currentEntry.getStoreTime();
-            delayedEntry.setStoreTime(currentStoreTime);
-        }
+        map.put(key, delayedEntry);
     }
 
     @Override
-    public DelayedEntry get(DelayedEntry entry) {
-        return queue.get(entry.getKey());
+    public void addFront(Collection<DelayedEntry> collection) {
+        if (isEmpty(collection)) {
+            return;
+        }
+        int expectedCapacity = map.size() + collection.size();
+        Map<Data, DelayedEntry> newMap = createMapWithExpectedCapacity(expectedCapacity);
```

Comment:
Because trying to insert failed store operations into the front of the map since it is `LinkedHashMap` and order is important. But also here is a room for improvement also and i will send a PR for it later.